### PR TITLE
Add Enum Types

### PR DESCRIPTION
**Pull request by d11r**
As @DanProgrammer1999 suggested, I added ENUM types to some of our types.

This should hopefully allow us to check the type during runtime.

Notice that some types, such as Literal and Nil don't have constructors so I couldn't implement the functionality for them.

Is this OK? How to proceed?

### CODE REVIEW

File: parser.h [hunk 1]
```diff
@@ -126,6 +173,7 @@ class Boolean : public Literal
 
 class Nil : public Literal
 {
+    // TODO: should we add type here?
```
Comment (DanProgrammer1999) [inline parser.h]: Yeah, just make constructor with no arguments which will assign the value of execType

**Review by DanProgrammer1999 — COMMENTED** [review]
Add getter in class `Element` for execType

File: parser.h (hunk 1 above)
Comment (d11r) [reply parser.h]: OK, I added this.

**Comment by d11r** [comment]
@DanProgrammer1999 I added the getter for execType in Element class

**Review by Picroc — APPROVED** [review]

**Pull request merged**